File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String[] sNumbers = new String[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            sNumbers[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(sNumbers, (a, b) -> (b + a).compareTo(a + b));
        
        // 가장 큰 수가 0인 경우
        if (sNumbers[0].equals("0")) return "0";
        
        StringBuilder answer = new StringBuilder();
        for (String s: sNumbers) {
            answer.append(s);
        }
        
        return answer.toString();
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    s_numbers = [str(n) for n in numbers] # 정수 전부 문자열 변환
    
    s_numbers.sort(key=lambda x: x*4, reverse=True)
    
    answer = ''.join(s_numbers)
    
    return '0' if answer[0] == '0' else answer